core/storage/user/gae_models.py

```diff
@@ -1130,8 +1130,17 @@ def apply_deletion_policy(cls, user_id: str) -> None:
         Args:
             user_id: str. The ID of the user whose data should be deleted.
         """
-        keys = cls.query(cls.creator_ids == user_id).fetch(keys_only=True)
-        datastore_services.delete_multi(keys)
+        user_subscriptions_models: List[UserSubscriptionsModel] = list(
+            cls.query(
+                cls.creator_ids == user_id
+            ).fetch()
+        )
+
+        for user_subscribers_model in user_subscriptions_models:
+            user_subscribers_model.creator_ids.remove(user_id)
+
+        cls.update_timestamps_multi(user_subscriptions_models)
+        cls.put_multi(user_subscriptions_models)
         cls.delete_by_id(user_id)
```

Comment:
I suggest putting a newline before this and a comment describing each of the two "sections" of this code, since otherwise it's a bit easy to miss this line at the end and realize it's not related to what comes above it.

Reply:
Done.